### ISSUE TITLE
Compiler gives error on r2d2::Pool<r2d2_mysql::MysqlConnectionManager>

### ISSUE THREAD

**Issue by shahmuratov** [outersky/r2d2-mysql#3]
Hi!
I'm trying to compile code:

```
struct Foo {
    pool: r2d2::Pool<r2d2_mysql::MysqlConnectionManager>,
} 
```

which gives me error

```
src/main.rs:65:5: 65:39 error: the trait bound `r2d2_mysql::MysqlConnectionManager: r2d2::ManageConnection` is not satisfied [E0277]
src/main.rs:65     pool: Pool<MysqlConnectionManager>,
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/main.rs:65:5: 65:39 help: run `rustc --explain E0277` to see a detailed explanation
src/main.rs:65:5: 65:39 note: required by `r2d2::Pool`
```

My Cargo.toml

```
[dependencies]
mysql = "*"
r2d2 = "*"
r2d2_mysql = "*"
rustc-serialize = "*"
```

If I try to use https://github.com/sfackler/r2d2-postgres and replace code with

```
struct Foo {
    pool: r2d2::Pool<r2d2_postgres::PostgresConnectionManager>,
} 
```

everything compiles.
I can't see real differences in this two cargos. Both have 

```
impl r2d2::ManageConnection for PostgresConnectionManager {
```

```
impl r2d2::ManageConnection for MysqlConnectionManager {
```

I would appreciate any help. Thanks

**Comment by shahmuratov** [comment]
So I figured it out. I've fixed version of r2d2 to 0.6.4 everything works. Hence I think problem in Error type (v0.7.0: \* Add an Error bound to ManageConnection::Error)

**Comment by shahmuratov** [comment]
Next problem which I faced is

```
let config = r2d2::Config::builder()
        .pool_size(100)
        .build();
    let manager = r2d2_mysql::MysqlConnectionManager::new("localhost:1234");

    let pool = Arc::new(r2d2::Pool::new(config, manager).unwrap());
```

doesn't compile.

```
src/main.rs:110:25: 110:40 error: the trait bound `std::result::Result<r2d2_mysql::MysqlConnectionManager, r2d2_mysql::Error>: r2d2::ManageConnection` is not satisfied [E0277]
src/main.rs:110     let pool = Arc::new(r2d2::Pool::new(config, manager).unwrap());
                                        ^~~~~~~~~~~~~~~
src/main.rs:110:25: 110:40 help: run `rustc --explain E0277` to see a detailed explanation
src/main.rs:110:25: 110:40 note: required by `r2d2::Pool::new`
```

**Comment by shahmuratov** [comment]
So the last problem was in my code and I fixed it, but main problem remains: error should implement std::error::Error trait. Current workaround is to fix version of r2d2 to 0.6.4

**Comment by outersky** [comment]
please try version 7.1.0 again ( already published to crate.io ), upgraded to the latest version of mysql and r2d2.
